pkg/vz/vm_darwin.go

```diff
@@ -334,6 +335,58 @@ func attachDisks(driver *driver.BaseDriver, vmConfig *vz.VirtualMachineConfigura
 	}
 	configurations = append(configurations, diffDisk)
 
+	for _, diskName := range driver.Yaml.AdditionalDisks {
+		d, err := store.InspectDisk(diskName)
+		if err != nil {
+			return fmt.Errorf("failed to run load disk %q: %q", diskName, err)
+		}
+
+		if d.Instance != "" {
+			return fmt.Errorf("failed to run attach disk %q, in use by instance %q", diskName, d.Instance)
+		}
+		logrus.Infof("Mounting disk %q on %q", diskName, d.MountPoint)
+		err = d.Lock(driver.Instance.Dir)
+		if err != nil {
+			return fmt.Errorf("failed to run lock disk %q: %q", diskName, err)
+		}
+		extraDiskPath := filepath.Join(d.Dir, filenames.DataDisk)
+
+		extraDiskFormat, err := imgutil.DetectFormat(extraDiskPath)
+		if err != nil {
+			return fmt.Errorf("failed to run detect disk format %q: %q", diskName, err)
+		}
+		if extraDiskFormat != "raw" {
+			rawPath := fmt.Sprintf("%s.raw", extraDiskPath)
+			qcow2Path := fmt.Sprintf("%s.qcow2", extraDiskPath)
+			if err = imgutil.QCOWToRaw(extraDiskPath, rawPath); err != nil {
+				return fmt.Errorf("failed to convert qcow2 disk %q to raw for vz driver: %w", diskName, err)
+			}
```

Comment:
Technically we don't know that the current format is qcow2, except for the fact that `limactl disk create` currently only creates either `raw` or `qcow2` formatted volumes. So this code could be kept more generic, like this:

```suggestion
			oldFormatPath := fmt.Sprintf("%s.%s", extraDiskPath, extraDiskFormat)
			if err = imgutil.ConvertToRaw(extraDiskPath, rawPath); err != nil {
				return fmt.Errorf("failed to convert %s disk %q to raw for vz driver: %w", extraDiskFormat, diskName, err)
			}
			...
```

Reply:
Good point, create a new issue to track this, I'll implement it after we get this merged if that's alright https://github.com/lima-vm/lima/issues/1438